### PR TITLE
fix: Use content header context for alerts

### PR DESCRIPTION
**Pull request by Al-Dani**
### Description

Alert context overrides button styles. It results in using incorrect color palette in light mode when alert is inside content layout header.
image

This PR creates `alert-header` context, which uses dark mode state for color tokens. Context is applied to alerts inside `content-header` context.

Related links, issue #, if available: AWSUI-21158

### How has this been tested?

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by jperals — APPROVED** [review]

File: src/alert/internal.tsx [hunk 1]
```diff
@@ -75,7 +75,7 @@ export default function InternalAlert({
       )}
       ref={mergedRef}
     >
-      <VisualContext contextName="alert">
+      <VisualContext contextName={'alert'}>
```
Comment (jperals) [inline src/alert/internal.tsx:78]: Nitpick: unnecessary change

**Review by jperals — DISMISSED** [review]

File: src/alert/internal.tsx (hunk 1 above)
Comment (Al-Dani) [reply src/alert/internal.tsx:78]: good catch!

**Review by jperals — APPROVED** [review]
